tests/integration/cli/tests/snapshot.rs

```diff
@@ -1268,3 +1268,12 @@ fn test_snapshot_quickjs() {
         .run_wasm(include_bytes!("./wasm/qjs.wasm"));
     assert_json_snapshot!(snapshot);
 }
+
+#[cfg_attr(any(target_env = "musl", target_os = "windows"), ignore)]
+#[test]
+fn test_snapshot_fs_rename() {
+    let snapshot = TestBuilder::new()
+        .with_name(function!())
+        .run_wasm(include_bytes!("./wasm/fs-rename.wasm"));
```

Comment:
I assume the test is written in Rust?

I'd say we take this opportunity to add a new `tests/rust-wasi-tests` crate, which as a bunch of `src/bin/xxx.rs` files for each `.wasm`. 

(We can then also add a `rust-wasix-tests` crate with wasix tests, etc. )

Reply:
This makes sense, I'll do it in this PR.

Reply:
We can later also add a make command that compiles all the tests and copies the wasm files to a well-known location, where they can be included from in tests.

Reply:
I just realized commenting "Done" on a conversation resolves it instead of adding a comment!

Reply:
@theduke it appears that the code in that new crate is being compiled by one of the tests, and since the APIs don't exist on Windows, it's failing. Any idea how we might fix that?

Reply:
I think we can just remove the crate from the workspace.

Reply:
let me try that.